### PR TITLE
replace statsd with prometheus client

### PR DESCRIPTION
**Pull request by AndrewChubatiuk**
## What type of PR is this? 
currently there's only one option to expose metrics - statsd client, which doesn't support custom metric name -> value separator
added metrics endpoint and ability to enable statsd output using prometheus-client

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## How is this tested?

- [ ] Unit tests (pytest, jest)
- [ ] E2E Tests (Cypress)
- [ ] Manually
- [ ] N/A

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -12,3 +12,4 @@ venv/
 /.github/
 /netlify.toml
 /setup/
+.github/
```
Comment (justinclift) [inline .dockerignore:15]: Accidental inclusion? :smile:

**Comment by justinclift** [comment]
I don't personally use either statsd nor Prometheus, so I don't have strong feelings with way. :smile:

@getredash/maintainers What do you reckon?

File: .dockerignore (hunk 1 above)
Comment (AndrewChubatiuk) [reply .dockerignore:15]: nope, .github is not needed inside container

**Comment by AndrewChubatiuk** [comment]
> I don't personally use either statsd nor Prometheus, so I don't have strong feelings with way. 😄
> 
> @getredash/maintainers What do you reckon?

cannot use this statsd library with victoriametrics. it has hardcoded colon as a separator, which is not configurable and looks like library is not well maintained

File: .dockerignore (hunk 1 above)
Comment (justinclift) [reply .dockerignore:15]: No worries. :smile:

**Comment by vtatarin** [comment]
Thank you for refactoring Job serializer and Athena config as well! This is a huge improvement, would love to see this merged!

**Comment by justinclift** [comment]
Uh oh, this is a very large PR. :frowning: 

Can it be broken up into smaller pieces that each individually work by themselves (aka not break things)?

That would make the review process be far more likely to go ok. :wink:

**Comment by wlach** [comment]
> Uh oh, this is a very large PR. 😦
> 
> Can it be broken up into smaller pieces that each individually work by themselves (aka not break things)?
> 
> That would make the review process be far more likely to go ok. 😉

I agree, it's difficult to give something this large a proper review to as-is. Even if everything in this PR is related to this feature (I'm not sure?), it looks at a glance like some changes (e.g. the frontend changes) could be landed seperately.

There's a number of tools that can be used to support a stacking workflow online: https://stacking.dev/stacking-site

**Comment by AndrewChubatiuk** [comment]
@wlach @justinclift 
divided all effort into PRs
https://github.com/getredash/redash/pull/6913
https://github.com/getredash/redash/pull/6914
https://github.com/getredash/redash/pull/6912

**Comment by justinclift** [comment]
@AndrewChubatiuk Cool, that seems like a good start. :smile:

**Comment by AndrewChubatiuk** [comment]
@wlach could you please review?

**Comment by AndrewChubatiuk** [comment]
@wlach @eradman  could you please review?

**Review by wlach — COMMENTED** [review]
I took a pass at this. This mostly looks reasonable to me, except for maybe adding a gotcha about statsd no longer being unconditionally enabled -- do we need to update any documentation or release notes?

I'm a little hestitant to hit the approve button just because I'm not super familiar with statsd or prometheus in the context of redash. I'd rather leave that to @eradman or someone else who has more domain context.

File: redash/settings/__init__.py [hunk 2]
```diff
@@ -22,9 +22,11 @@
 REDIS_URL = add_decode_responses_to_redis_url(_REDIS_URL)
 PROXIES_COUNT = int(os.environ.get("REDASH_PROXIES_COUNT", "1"))
 
+STATSD_ENABLED = parse_boolean(os.environ.get("REDASH_STATSD_ENABLED", "false"))
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply redash/settings/__init__.py:25]: it makes no sense to enable statsd by default if it has no remote endpoint configured

[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply redash/settings/__init__.py:25]: statsd uses push model, and prometheus - pull. for statsd you should specify remote url to push metrics to and prometheus exposes metrics at configured endpoint

[the remark this answers is not in the document]
Comment (wlach) [reply redash/settings/__init__.py:25]: Got it, makes sense!

File: tests/metrics/test_database.py [hunk 3]
```diff
@@ -3,8 +3,8 @@
...
 
 
-@patch("statsd.StatsClient.timing")
+@patch("prometheus_client.Histogram.observe")
 class TestDatabaseMetrics(BaseTestCase):
-    def test_db_request_records_statsd_metrics(self, timing):
+    def test_db_request_records_metrics(self, observe):
         self.factory.create_query()
-        timing.assert_called_with("db.changes.insert", ANY)
+        observe.assert_called_with(ANY)
```
[the remark this answers is not in the document]
Comment (AndrewChubatiuk) [reply tests/metrics/test_database.py:10]: interface has changed and observe function calculates histograms and expect time at the input. we cannot predict the time it takes to run a function, so ANY value was set here. regarding metric name, it should not be passed anymore as it's initialized in a different manner from statsd python client

**Pull request closed without merging**